src/extension/features/budget/custom-average-budgeting/index.js

```diff
@@ -0,0 +1,122 @@
+import { controllerLookup, getEmberView } from 'toolkit/extension/utils/ember';
+import {
+  getEntityManager,
+  isCurrentMonthSelected,
+  isCurrentRouteBudgetPage,
+} from 'toolkit/extension/utils/ynab';
+import { formatCurrency } from 'toolkit/extension/utils/currency';
+import { Feature } from '../../feature';
+
+export class CustomAverageBudgeting extends Feature {
+  constructor() {
+    super();
+    this.timeframe = this.settings.enabled;
+    this.transactions = getEntityManager().transactionsCollection._internalDataArray;
+    this.lookbackFrame = this._calculateLookback();
+    this.selectedCategory = this._getSelectedCategoryId();
+    this.average = 0;
+  }
+
+  shouldInvoke() {
+    return isCurrentRouteBudgetPage() && isCurrentMonthSelected();
+  }
+
+  _calculateLookback() {
+    var today = new Date();
+    var d;
+    var month;
+    var year;
+    var arr = [];
+
+    for (var i = this.timeframe; i >= 1; i -= 1) {
+      d = new Date(today.getFullYear(), today.getMonth() - i, 1);
+      month = d.getMonth();
+      year = d.getFullYear();
+      arr.push(`${month}|${year}`);
+    }
+
+    return arr.join('`');
+  }
+
+  _calculateAverage() {
+    var timeframe = this.lookbackFrame.split('`');
+    var filteredTransactions = this.transactions.filter((transaction) => {
+      var month = transaction.date._internalUTCMoment._i[1];
+      var year = transaction.date._internalUTCMoment._i[0];
+      return (
+        timeframe.includes(`${month}|${year}`) &&
```

Comment:
transactions just use `moment` underneath so you can do: `transaction.date.format('YYYY|MM')` rather than concatenating yourself